lib/iris/tests/unit/analysis/test_Aggregator.py

```diff
@@ -315,6 +319,37 @@ def test_kwarg_pass_through_combined_kwargs(self):
         expected_kwargs.update(call_kwargs)
         lazy_func.assert_called_once_with(data, axis, **expected_kwargs)
 
+class Test_aggregated_by(tests.IrisTest):
+    def setUp(self):
+        # This sets up a cube with a time coordinate containing two years and
+        # two months.
+        # This will test the ability of the aggregator to correctly calculate
+        # the order of months in the new coordinate array.
+        latitude = DimCoord(np.linspace(-90, 90, 4),
+                            standard_name='latitude',
+                            units='degrees')
+        longitude = DimCoord(np.linspace(45, 360, 8),
+                             standard_name='longitude',
+                             units='degrees')
+        time_unit = unit.Unit('days since epoch',
```

Comment:
Thanks AJ, I've done that quite a lot.  I was unsure whether the test would still work when setting an explicit date because that's how I first wrote it and it didn't work, but I've been through so many iterations of this test trying to achieve the failure mode that I was unsure which bits of syntax had which effect.

Anyway, I have added the explicit date and tried to make it a little more clear what the point of the test is.  What do you think, any better?